src/Json.Schema.ToDotNet.UnitTests/CommandLineTests.cs

```diff
@@ -0,0 +1,55 @@
+﻿// Copyright (c) Microsoft Corporation.  All Rights Reserved.
+// Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Linq;
+using CommandLine;
+using FluentAssertions;
+using Microsoft.Json.Schema.ToDotNet.CommandLine;
+using Xunit;
+
+namespace Microsoft.Json.Schema.ToDotNet.UnitTests
+{
+    public class CommandLineTests
+    {
+        [Fact]
+        public void CorrectlyParseGenerateIntegerAs()
+        {
+            (string argsString, GenerateIntegerOption? expectedGenerateIntegerAs, string expectedErrorParameter)[] testCases = new[]
+            {
+                ("--schema-name Sarif --schema-file-path \"C:\\sarif.json\" --output-directory \"C:\\Autogenerated\" --namespace-name Microsoft.CodeAnalysis.Sarif --root-class-name SarifLog", (GenerateIntegerOption?)GenerateIntegerOption.Int, null),
+                ("--schema-name Sarif --schema-file-path \"C:\\sarif.json\" --output-directory \"C:\\Autogenerated\" --namespace-name Microsoft.CodeAnalysis.Sarif --root-class-name SarifLog --generate-integer-as=Int", (GenerateIntegerOption?)GenerateIntegerOption.Int, null),
+                ("--schema-name Sarif --schema-file-path \"C:\\sarif.json\" --output-directory \"C:\\Autogenerated\" --namespace-name Microsoft.CodeAnalysis.Sarif --root-class-name SarifLog --generate-integer-as=int", (GenerateIntegerOption?)GenerateIntegerOption.Int, null),
```

Comment:
What are there case variations for int type and not the others?  Is there a more dynamic way you can generate these test inputs since the majority of the string is the same and at the same time have it generate all the case variants if you are looking to support case insensitivity

Reply:
Did some reflector and removed the duplication and simplified the input.
The reason the case sensitive test does not cover all is I was just to simply verify the case insensitive flag is set, case insensitive is build-in option for the ConmandLine package. Let me know I can generate all possible case combination of all the cases if we like.